feat(event-gateway): improve schema validation docs

jakubdyszkiewicz opened the pull request:
## Description

A bunch of changes to schema validation docs
* Add failure modes and explain what are the headers added by mark
* Add an observability section to figure out how to troubleshoot if anything fails
* I find expressions like "Validate all topics" not accurate, we validate messages on the topics.
* Fixed a couple of copy-paste errors between produce and consume. 

### Sending message to schema
* Fixed the misconception that we send a message to the schema registry to validate it. We only pull the schema from the schema registry and cache it.
* Fixed diagrams.
* Changed "event broker" to "kafka broker". We use "Kafka broker" in docs, but "Event broker" in diagrams for some reason.
* I believe client should be always on the left in diagrams. It's hard to parse consume diagrams when we switch the order of columns

### Use cases
* Suffix is very rarely used, if we want to select subset of topics, it should be by prefix
* Key validation will be rarely used, it should not be added by default. I only left it when `mark` is used
* "Validate all messages for a topic", but then we select subset of topics. I changed the use cases so that the first selects all,the  second selects subset, third selects one specific topic.
* I find this common use case table to be hard to read because it tries to share use cases for produce and consume, so the descriptions are a bit lacking. I split this for consume and produce and moved descriptions 1:1.
* "inferred JSON schema" is not correct. We just check if object is a valid json. Inferred JSON schema suggest there is actual JSON schema in place. 

I'm happy to provide further explanation if needed.

## Preview Links


## Checklist 

- [X] Tested how-to docs. If not, note why here. 
- [X] All pages contain metadata.
- [X] Any new docs link to existing docs.
- [X] All autogenerated instructions render correctly (API, decK, Konnect, Kong Manager).
- [X] Style guide (capitalized gateway entities, placeholder URLs) implemented correctly.
- [X] Every page has a `description` entry in frontmatter.
- [X] Add new pages to the product documentation index (if applicable).

jakubdyszkiewicz commented on app/_event_gateway_policies/schema-validation-consume/examples/validate-all-messages-for-topic.yml:
```diff
@@ -1,6 +1,6 @@
-title: Validate all messages for a topic
```
I can restore it, but I find "all" redundant.

fabianrbz commented on app/_event_gateway_policies/schema-validation-consume/examples/validate-all-topics-confluent.yml:
```diff
@@ -1,12 +1,12 @@
+title: Validate messages against a Confluent Schema Registry
 
-description: Ensure that every consumed topic is validated against a Confluent Schema Registry, and mark any messages that don't conform.
+description: Ensure that all messages consumed from any topic are validated against a Conflent Schema Registry, and mark messages that don't conform.
```
```suggestion
description: Ensure that all messages consumed from any topic are validated against a Confluent Schema Registry, and mark messages that don't conform.
```

jakubdyszkiewicz commented:
```
found 0 vulnerabilities
Missing redirects:
- Renamed app/_event_gateway_policies/schema-validation-consume/examples/validate-all-messages-for-topic.yml, url: /event-gateway/policies/schema-validation-consume/examples/validate-all-messages-for-topic/
- Renamed app/_event_gateway_policies/schema-validation-produce/examples/validate-all-topics-confluent.yml, url: /event-gateway/policies/schema-validation-produce/examples/validate-all-topics-confluent/
- Deleted app/_event_gateway_policies/schema-validation-consume/examples/validate-all-topics-json.yml, url: /event-gateway/policies/schema-validation-consume/examples/validate-all-topics-json/
- Renamed app/_event_gateway_policies/schema-validation-produce/examples/validate-all-messages-for-topic.yml, url: /event-gateway/policies/schema-validation-produce/examples/validate-all-messages-for-topic/
- Renamed app/_event_gateway_policies/schema-validation-consume/examples/validate-all-topics-confluent.yml, url: /event-gateway/policies/schema-validation-consume/examples/validate-all-topics-confluent/
- Deleted app/_event_gateway_policies/schema-validation-produce/examples/validate-all-topics-json.yml, url: /event-gateway/policies/schema-validation-produce/examples/validate-all-topics-json/
```

How do I fix this?

fabianrbz commented:
@jakubdyszkiewicz you can ignore that check, it's not working right, a fixed was pushed to main, we need to update the release branch
